cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/analyzer/ReplicationFactorChangeTest.java

```diff
@@ -270,6 +279,21 @@ private void prepareContext() {
                                  Class<? extends Throwable> exceptionClass,
                                  ClusterModel clusterModel,
                                  Boolean expectedToOptimize) throws Exception {
-    return new Object[]{tid, topics, replicationFactor, goal(goalClass), exceptionClass, clusterModel, expectedToOptimize};
+    Goal goal;
+    if (goalClass == MinTopicLeadersPerBrokerGoal.class) {
```

Comment:
In this test class, does it matter for the other goals if we make the config changes proposed under this if case? -- i.e. is it possible to drop if-else and just do the config changes regardless of what the goal is?

Reply:
The `if-else` statement can be dropped. However, I feel like it affects the readability since readers might be wondering why some config props such as `MIN_TOPIC_LEADERS_PER_BROKER_CONFIG` and `TOPICS_WITH_MIN_LEADERS_PER_BROKER_CONFIG` is required for this testing.

Whereas with this `if-else` statement, it's very clear that these config props only need to be set for the `MinTopicLeadersPerBrokerGoal`.

I will drop the `if-else` statement for now